core/components/babel/src/Plugins/Events/OnContextSave.php

```diff
@@ -10,15 +10,54 @@
 
 class OnContextSave extends Plugin
 {
-    /**
+     /**
      * Refresh the babel cache
      * @return void
      */
     public function process()
     {
+        $context = &$this->scriptProperties['context'];
+        $this->syncSetting($context);
+        
         $cacheManager = $this->modx->getCacheManager();
         $cacheManager->refresh([
             'babel' => [],
         ]);
     }
+    
+    private function syncSetting($currentCtx) {
+        if(!$currentCtx) return;
+        $keysToGroup = $this->babel->contextKeyToGroup[$currentCtx->get('key')];
+        
+        foreach($keysToGroup as $context_key) {
+            if($context_key == $currentCtx->get('key')) continue;
+            foreach($this->babel->config['syncOptions'] as $key) {
+                $currentCtxSetting = $this->modx->getObject('modContextSetting', [
+                    'context_key' => $currentCtx->get('key'),
+                    'key' => $key,
+                ]);
+                if(!$currentCtxSetting) continue;
+                
+                $ctxSetting = $this->modx->getObject('modContextSetting', [
+                    'context_key' => $context_key,
+                    'key' => $key,
+                ]);
+                if(!$ctxSetting) {
+                    
+                    $ctxSetting = $this->modx->newObject('modContextSetting', [
```

Comment:
It helps to use fromArray and enable the setPrimaryKeys option.

Reply:
See: https://github.com/Jako/Babel/blob/3.5.0/core/components/babel/src/Plugins/Events/OnContextSave.php#L63-L71